### PR TITLE
Polygon tests

### PR DESCRIPTION
**Pull request by oli-newcombe-faculty**
Added tests for PolygonMatcher class.

 - Testing polygon edge cases
 - Testing area calculations
 - Testing alignment logic for example polygons
 - Added fixtures for example polygons
 - Add filtering functionality and corresponding tests.

### CODE REVIEW

File: tests/test_polygons.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+from geopandas import GeoDataFrame
+from shapely import Polygon, unary_union
+
+from data_quality_utils.polygon.polygon_matcher import PolygonMatcher
+
+BASE_CRS = "EPSG:4326"
+MERCATOR_CRS = "EPSG:3857"
+
+
+def create_square_coords(lat: int = 0, long: int = 0, add_buffer=0.01) -> Polygon:
```
Comment (simona-lihosithova-faculty) [inline tests/test_polygons.py]: maybe add type hint for `add_buffer` for consistency

File: tests/test_polygons.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+
+@pytest.fixture
+def square_df() -> GeoDataFrame:
+    base_list = create_square_coords()
+    return create_polygon_df(base_list, BASE_CRS)
+
+
+@pytest.fixture
+def small_square_df(add_buffer=0.001) -> GeoDataFrame:
```
Comment (simona-lihosithova-faculty) [inline tests/test_polygons.py]: same here

File: tests/test_polygons.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+    assert sum(diff_df.area) == large_area_total
+
+
+def test_area_proportions(
+    rectangle_df: GeoDataFrame,
+    triangle_df: GeoDataFrame,
+    shorter_pentagon_df: GeoDataFrame,
+):
+    matcher = PolygonMatcher()
```
Comment (simona-lihosithova-faculty) [inline tests/test_polygons.py:221]: could you add a short comment stating what each case is testing?

**Comment by jonholdship-faculty** [comment]
Please add a test that the large_area_discrepancy_proportion returns the correct value for some inputs with a known ratio. If you do that and address Simona's comments, I'm happy for you to merge.
